Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp

```diff
@@ -782,6 +782,8 @@ void NetworkConnectionToWebProcess::registerURLSchemesAsCORSEnabled(Vector<Strin
 
 void NetworkConnectionToWebProcess::cookiesForDOM(const URL& firstParty, const SameSiteInfo& sameSiteInfo, const URL& url, FrameIdentifier frameID, PageIdentifier pageID, IncludeSecureCookies includeSecureCookies, ShouldAskITP shouldAskITP, ShouldRelaxThirdPartyCookieBlocking shouldRelaxThirdPartyCookieBlocking, CompletionHandler<void(String cookieString, bool secureCookiesAccessed)>&& completionHandler)
 {
+    if (!m_networkProcess->allowsFirstPartyForCookies(m_webProcessIdentifier, firstParty))
```

Comment:
Why aren't we doing MESSAGE_CHECKs() ? Seems like we'd want to kill the WebProcess sending the bad IPC since we would consider it as compromised. This is what we usually do. I'm unclear why we're not doing the same here.